crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -233,7 +233,9 @@ impl PgManager {
             .or(before)
             .map(|cursor| self.parse_obj_cursor(&cursor))
             .transpose()?;
-
+        let coin_type = parse_to_type_tag(Some(coin_type))
+            .map_err(|_| Error::Internal("Invalid coin type.".to_string()))?
```

Comment:
Can you propagate the actual error through?

Reply:
Ah wait yeah, this should typically be a client error, +1